fastcache.go

```diff
@@ -121,6 +122,26 @@ func (c *Cache) UpdateStats(s *Stats) {
 	}
 }
 
+// Keys retrieves all cached keys matching regex pattern
+func (c *Cache) Keys(pattern string) (keys [][]byte, err error) {
```

Comment:
Method with such a signature will blow up on caches containing billions of matching keys, since it have to allocate memory and pointers for all the returned keys.

It would be better to have a method for iterating over all the available entries like the following:
```go
// VisitAllEntries calls f for all the cache entries.
//
// The function returns immediately if f returns non-nil error.
// It returns the given error.
//
// f cannot hold pointers to k and v contents after returning.
func (c *Cache) VisitAllEntries(f func(k, v []byte) error) error
```

Then this method could be used for keys scanning with the given regexp:
```go
func Keys(c *Cache, r *regexp.Regexp) []string {
    var keys []string
    c.VisitAllEntries(func(k, v []byte) error {
        if r.Match(k) {
            keys = append(keys, string(k))
        }
        return nil
    })
    return keys
}
```

But I'm skeptical for all this stuff. What is the purpose of it?

Reply:
The need is to find all grouped items, like finding all cached users, if the cache key has a pattern like "user:<id>". It's a situation I'm dealing with and I'm about to switch to an in-memory cache, and I was considering that maybe this package could use such a feature.

Obviously, I don't understand yet your package entirely and I have to dig more. No problem if this idea doesn't meet your view on the whole package, we can close the PR. :)